File: clients/swift/src/RateLimiting/RetryPolicy.swift
import Foundation

/// Controls how the client handles 429 rate-limit responses.
public struct RetryPolicy: Sendable {
    /// Maximum number of retry attempts after a 429. Default: 3.
    public var maxRetries: Int
    /// Extra time added on top of the server's reset window. Default: 1 second.
    public var buffer: TimeInterval

    public init(maxRetries: Int = 3, buffer: TimeInterval = 1) {
        self.maxRetries = maxRetries
        self.buffer = buffer
    }

    public static let `default` = RetryPolicy()
}

File: clients/csharp/src/RateLimiting/RetryPolicy.cs
namespace MyMind.RateLimiting;

/// <summary>Controls how the client handles 429 rate-limit responses.</summary>
public class RetryPolicy
{
    /// <summary>Maximum number of retry attempts after a 429. Default: 3.</summary>
    public int MaxRetries { get; init; } = 3;

    /// <summary>Extra time added on top of the server's reset window. Default: 1 second.</summary>
    public TimeSpan BufferTime { get; init; } = TimeSpan.FromSeconds(1);

    public static readonly RetryPolicy Default = new();
}
